src/Type/ObjectType/RootQuery.php

```diff
@@ -790,10 +803,16 @@ function ( $post ) use ( $post_type_object ) {
 						return $context->get_loader( 'post' )->load_deferred( $post_id )->then(
 							function ( $post ) use ( $post_type_object ) {
 
+								// if the post type object isn't an instance of WP_Post_Type, return
 								if ( ! $post_type_object instanceof \WP_Post_Type ) {
 									return null;
 								}
 
+								// if the post isn't an instance of a WP_Post or Post model, return
+								if ( ! $post instanceof \WP_Post && ! $post instanceof Post ) {
```

Comment:
See above re the return type, if were accepting both a `WP_Post` and a `Post` model.